### PR TITLE
fixed cross cluster searching resolves #692

### LINKED ISSUE #692: elasticsearch reader doesn't work with cross cluster search
It appears cross cluster search can't answer the request for index settings so if you try to run a job against the search nodes you just get ```index specified in reader does not exist``` and the job immediately exits.

### PR DESCRIPTION
**Pull request by jsnoble**
make sure to specify the correct connection on the job that it points to the node

### CODE REVIEW

File: lib/readers/elasticsearch_date_range/slicer.js [hunk 1]
```diff
@@ -23,6 +23,16 @@ function newSlicer(context, opConfig, executionContext, retryData, logger, clien
...
+            // regex searches for zero or more chars preceding a '*' , zero or more chars after
+            // before a ':' then any amount of chars before and after another '*'
+            // =>  ie  es_d*:2017-domains*
+            const wildCardRegex = RegExp(/\*/g);
+            const isWildCardRegexSearch = opConfig.index.match(wildCardRegex);
+            // We cannot reliable search
+            if (isWildCardRegexSearch !== null) {
+                logger.warn(`Running a regex or cross cluster search for ${opConfig.index}, there is no reliable way to verify index and max_result_window`);
+                return Promise.resolve(true);
+            }
```
Comment (kstaken) [inline lib/readers/elasticsearch_date_range/slicer.js]: Shouldn't this be in elasticsearch_api with the rest of the logic related to index validation?

**Comment by kstaken** [comment]
Unit tests are failing

**Comment by jsnoble** [comment]
needs to be used in conjunction with https://github.com/terascope/elasticsearch_api/pull/12

**Comment by kstaken** [comment]
unit and integration tests pass. Manual xsearch job works.

**Pull request merged**